miro_preprocessor/main.tf

```diff
@@ -1,56 +1,24 @@
 module "xml_to_json_converter" {
-  source        = "../terraform/ecs_script_task"
-  task_name     = "xml_to_json_converter"
-  app_uri       = "${module.ecr_repository_xml_to_json_converter.repository_url}:${var.release_ids["xml_to_json_converter"]}"
-  task_role_arn = "${module.ecs_xml_to_json_converter_iam.task_role_arn}"
+  source = "xml_to_json_converter"
 
-  env_vars = [
-    "{\"name\": \"BUCKET\", \"value\": \"${data.terraform_remote_state.platform.bucket_miro_data_id}\"}",
-    "{\"name\": \"AWS_DEFAULT_REGION\", \"value\": \"${var.aws_region}\"}",
-  ]
-}
-
-module "ecs_xml_to_json_converter_iam" {
-  source = "../terraform/ecs_iam"
-  name   = "xml_to_json_converter"
-}
-
-module "ecr_repository_xml_to_json_converter" {
-  source = "../terraform/ecr"
-  name   = "xml_to_json_converter"
-}
-
-resource "aws_iam_role_policy" "xml_to_json_converter_read_from_s3" {
-  role   = "${module.ecs_xml_to_json_converter_iam.task_role_name}"
-  policy = "${data.aws_iam_policy_document.s3_read_miro_data.json}"
-}
+  bucket_miro_data_id = "${data.terraform_remote_state.platform.bucket_miro_data_id}"
+  release_ids         = "${var.release_ids}"
 
-resource "aws_iam_role_policy" "xml_to_json_converter_read_from_s3" {
-  role   = "${module.ecs_xml_to_json_converter_iam.task_role_name}"
-  policy = "${data.aws_iam_policy_document.s3_write_miro_data.json}"
+  s3_read_miro_data_json  = "${data.aws_iam_policy_document.s3_read_miro_data.json}"
+  s3_write_miro_data_json = "${data.aws_iam_policy_document.s3_write_miro_data.json}"
 }
 
-data "aws_iam_policy_document" "s3_read_miro_data" {
-  statement {
-    actions = [
-      "s3:Get*",
-      "s3:List*",
-    ]
+module "xml_to_json_run_task" {
+  source = "xml_to_json_run_task"
 
-    resources = [
-      "${data.terraform_remote_state.platform.bucket_miro_data_arn}/source",
-    ]
-  }
-}
+  bucket_miro_data_id    = "${data.terraform_remote_state.platform.bucket_miro_data_id}"
+  bucket_miro_data_arn   = "${data.terraform_remote_state.platform.bucket_miro_data_arn}"
+  lambda_error_alarm_arn = "${data.terraform_remote_state.lambda.lambda_error_alarm_arn}"
 
-data "aws_iam_policy_document" "s3_write_miro_data" {
-  statement {
-    actions = [
-      "s3:Put*",
-    ]
+  s3_read_miro_data_json = "${data.aws_iam_policy_document.s3_read_miro_data.json}"
 
-    resources = [
-      "${data.terraform_remote_state.platform.bucket_miro_data_arn}/json",
-    ]
-  }
+  task_defintion_arn = "${module.xml_to_json_converter.task_arn}"
```

Comment:
`task_defintion_arn`?

`definition`?

Reply:
Good point - task_arn actually has a different meaning in AWS as the `arn` of a running instance of a task definition, as distinct from the definition itself.

Reply:
In case it's not clear - this should be `task_defintion_arn`.